File: pyg_shop_web/src/main/webapp/js/service/goodsService.js
//服务层
app.service('goodsService', function ($http) {

    //读取列表数据绑定到表单中
    this.findAll = function () {
        return $http.get('../goods/findAll');
    };
    //分页
    this.findPage = function (page, rows) {
        return $http.get('../goods/findPage/' + page + '/' + rows);
    };
    //查询实体
    this.findOne = function (id) {
        return $http.get('../goods/findOne/' + id);
    };
    //增加
    this.add = function (entity) {
        return $http.post('../goods/add', entity);
    };
    //修改
    this.update = function (entity) {
        return $http.post('../goods/update', entity);
    };
    //删除
    this.dele = function (ids) {
        return $http.get('../goods/delete/' + ids);
    };
    //搜索
    this.search = function (page, rows, searchEntity) {
        return $http.post('../goods/search/' + page + "/" + rows, searchEntity);
    }
});

File: pyg_sellergoods_interface/src/main/java/com/pinyougou/sellergoods/service/GoodsService.java
package com.pinyougou.sellergoods.service;

import com.pinyougou.pojo.TbGoods;
import com.pinyougou.vo.Goods;
import entity.PageResult;

import java.util.List;

/**
 * 服务层接口
 *
 * @author Administrator
 */
public interface GoodsService {

    /**
     * 返回全部列表
     *
     * @return
     */
    List<TbGoods> findAll();


    /**
     * 返回分页列表
     *
     * @return
     */
    PageResult findPage(int pageNum, int pageSize);


    /**
     * 增加
     */
    void add(Goods goods);


    /**
     * 修改
     */
    void update(TbGoods goods);


    /**
     * 根据ID获取实体
     *
     * @param id
     * @return
     */
    TbGoods findOne(Long id);


    /**
     * 批量删除
     *
     * @param ids
     */
    void delete(Long[] ids);

    /**
     * 分页
     *
     * @param pageNum  当前页 码
     * @param pageSize 每页记录数
     * @return
     */
    PageResult findPage(TbGoods goods, int pageNum, int pageSize);

    void goodsAudit(Long[] selectIds, String auditStatus);

    void deleteGoods(Long[] selectIds);

    void marketGoods(Long[] selectIds, String ismarketable);
}
